url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+		try {
+			int _type = SEPTEMBER;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:323:11: ( 'september' ( 's' )? | 'sep' ( DOT )? | 'sept' ( DOT )? )
+			int alt29=3;
+			int LA29_0 = input.LA(1);
+			if ( (LA29_0=='s') ) {
+				int LA29_1 = input.LA(2);
+				if ( (LA29_1=='e') ) {
+					int LA29_2 = input.LA(3);
+					if ( (LA29_2=='p') ) {
+						int LA29_3 = input.LA(4);
+						if ( (LA29_3=='t') ) {
+							int LA29_4 = input.LA(5);
+							if ( (LA29_4=='e') ) {
+								alt29=1;
+							}
+
+							else {
+								alt29=3;
+							}
+
+						}
+
+						else {
+							alt29=2;
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 29, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 29, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 29, 0, input);
+				throw nvae;
+			}
+
+			switch (alt29) {
+				case 1 :
+					// HiveLexer.g:323:13: 'september' ( 's' )?
+					{
+					match("september"); 
+
+					// HiveLexer.g:323:25: ( 's' )?
+					int alt26=2;
+					int LA26_0 = input.LA(1);
+					if ( (LA26_0=='s') ) {
+						alt26=1;
+					}
+					switch (alt26) {
+						case 1 :
+							// HiveLexer.g:323:25: 's'
+							{
+							match('s'); 
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:323:33: 'sep' ( DOT )?
+					{
+					match("sep"); 
+
+					// HiveLexer.g:323:39: ( DOT )?
+					int alt27=2;
+					int LA27_0 = input.LA(1);
+					if ( (LA27_0=='.') ) {
+						alt27=1;
+					}
+					switch (alt27) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 3 :
+					// HiveLexer.g:323:46: 'sept' ( DOT )?
+					{
+					match("sept"); 
+
+					// HiveLexer.g:323:53: ( DOT )?
+					int alt28=2;
+					int LA28_0 = input.LA(1);
+					if ( (LA28_0=='.') ) {
+						alt28=1;
+					}
+					switch (alt28) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 7198:7283, 7733:7818, 7893:7978, 8056:8141, 8216:8301, 8612:8697.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.